src/providers/bedrock/utils.ts

```diff
@@ -157,47 +158,236 @@ export const transformAI21AdditionalModelRequestFields = (
   return additionalModelRequestFields;
 };
 
-export async function getAssumedRoleCredentials(
+async function assumeRoleWithWebIdentity(token: string, roleArn: string) {
+  const params = new URLSearchParams({
+    Version: '2011-06-15',
+    Action: 'AssumeRoleWithWebIdentity',
+    RoleArn: roleArn,
+    RoleSessionName: `eks-${Date.now()}`,
+    WebIdentityToken: token,
+  });
+
+  const response = await fetch('https://sts.amazonaws.com', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: params.toString(),
+  });
+
+  if (!response.ok) {
+    const errorMessage = await response.text();
+    console.error({ message: `STS error ${errorMessage}` });
+    throw new Error(`STS request failed: ${response.status}`);
+  }
+
+  const data = await response.text();
+  return parseXml(data);
+}
+
+async function getAssumedWebIdentityCredentials(
   c: Context,
   awsRoleArn: string,
   awsExternalId: string,
-  awsRegion: string,
-  creds?: {
-    accessKeyId: string;
-    secretAccessKey: string;
-    sessionToken?: string;
-  }
+  awsRegion: string
 ) {
-  const cacheKey = `${awsRoleArn}/${awsExternalId}/${awsRegion}`;
   const getFromCacheByKey = c.get('getFromCacheByKey');
   const putInCacheWithValue = c.get('putInCacheWithValue');
 
+  if (process.env.AWS_WEB_IDENTITY_TOKEN_FILE && process.env.AWS_ROLE_ARN) {
```

Comment:
Can we please use env(c) instead of process.env for this. This might break in workerd runtime

Reply:
done